### PR TITLE
Task 41.1: BE: Implement API for location

### PR DESCRIPTION
**Pull request by di-kinosian**

### CODE REVIEW

File: src/location/location.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { firstValueFrom } from 'rxjs';
+import { ErrorMessages } from 'src/main/constants/messages.constants';
+
+@Injectable()
+export class LocationService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getCountries() {
+    try {
+      const response = await firstValueFrom(
```
Comment (VYaniv) [inline src/location/location.service.ts:13]: Do you know what it do `firstValueFrom` ?

File: src/location/location.service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      return response.data.data;
+    } catch (error) {
+      throw new HttpException(
+        error.response?.data || ErrorMessages.FAILED_FETCH_COUNTRIES,
+        error.response?.status || HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  async getCitiesByCountry(country: string) {
```
Comment (VYaniv) [inline src/location/location.service.ts]: Returned type

File: src/location/location.service.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { HttpService } from '@nestjs/axios';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { firstValueFrom } from 'rxjs';
+import { ErrorMessages } from 'src/main/constants/messages.constants';
+
+@Injectable()
+export class LocationService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getCountries() {
```
Comment (VYaniv) [inline src/location/location.service.ts]: Returned type

File: src/location/location.service.ts (hunk 1 above)
Comment (di-kinosian) [reply src/location/location.service.ts:13]: @VYaniv, the firstValueFrom method converts an Observable to a Promise. If I understand correctly, it allows us to work with the response as a Promise and use async/await workflows.

File: src/location/location.service.ts (hunk 3 above)
Comment (di-kinosian) [reply src/location/location.service.ts]: done

File: src/location/location.service.ts (hunk 1 above)
Comment (VYaniv) [reply src/location/location.service.ts:13]: Thanks

**Review by VYaniv — APPROVED** [review]

File: src/location/location.service.ts (hunk 2 above)
Comment (di-kinosian) [reply src/location/location.service.ts]: done

**Pull request merged**